### PR TITLE
LinkedIn: Migrate to the new posts API

### PR DESCRIPTION
**Pull request by jnv**
## TODO

- [x] Migrate posts map
- [x] Migrate publish post (text-only)
- [x] Publish post with images upload
- [x] Posts map with images' URLs
- [ ] ~Video uploads?~ – not covered by the original map so I'll ignore it for now

## Description

This migrates relevant LinkedIn maps to the new Posts API. However, there are some missing functionality due to API changes:

- In `social-media/posts` URLs to image and video attachments are missing, since the new API doesn't support projections; this is fixable by doing additional requests
- In `social-media/publish-post`, publishing a link will omit the link preview

The second issue is particularly problematic. [Per docs](https://learn.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/posts-api?view=li-lms-2023-02&tabs=curl#article-post-creation-sample-request):

> Posts API does not support url scraping for article post creation as it introduces level of unpredictability in how a post is going to look when API partners create it. Instead, API partners need to set article fields such as thumbnail, title and description within the post when creating an article post.

**This currently cannot be fixed.** To provide the same experience we'd have to scrape the URL on our own, but that's not possible with the security limitations of SDK, and the current profile doesn't accommodate for passing link details externally.

## Motivation and Context

The currently used ugcPosts and share APIs are deprecated and will be disabled on Feb 28: https://www.linkedin.com/developers/news/featured-updates/versioning-content-launch

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: grid/social-media/posts/maps/__snapshots__/linkedin.test.ts.snap [hunk 1]
```diff
@@ -20,169 +20,142 @@ Ok {
...
         "attachments": Array [
           Object {
-            "altText": undefined,
-            "description": undefined,
-            "duration": undefined,
-            "height": undefined,
-            "preview": "http://jrs-stage.int.railwaymen.org/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBa3U2IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--edd7dcfe90f218434c9f632d5767e04a96531f4d/PI-PNL-02-0.jpg",
-            "title": "1 test estates link 1",
-            "type": "link",
-            "url": "http://jrs-stage.int.railwaymen.org/content/8f36d0af-bcc8-4294-8708-bd6baab80424",
```
Comment (janhalama) [inline grid/social-media/posts/maps/__snapshots__/linkedin.test.ts.snap:30]: note: Missing URL and preview URL of images in attachments makes the profile much less usable.

**Review by janhalama — COMMENTED** [review]
I went through the PR, LinkedIn API is hard and it is even harder to do code review without reading the API docs. I have just two notes.

> publishing a link will omit the link preview -> This currently cannot be fixed. To provide the same experience we'd have to scrape the URL on our own, but that's not possible with the security limitations of SDK, and the current profile doesn't accommodate for passing link details externally.

We should come with new profile version to support it.

File: grid/social-media/posts/maps/__snapshots__/linkedin.test.ts.snap (hunk 1 above)
Comment (jnv) [reply grid/social-media/posts/maps/__snapshots__/linkedin.test.ts.snap:30]: Yup, this is actually holding me back from making this PR ready for review. I'm exploring how to fix it.

File: grid/social-media/publish-post/maps/linkedin.suma [hunk 2]
```diff
@@ -8,50 +8,64 @@ For text-only posts, links, and images (up to 200MB per image).
...
-    shareMediaCategory = "IMAGE"
+    content = undefined // FIXME: Avoid SDK's "fun" feature of merging objects implicitly
   }
-
-  // https://docs.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/share-api#post-shares
-  http POST "/v2/shares" {
+  set if (Array.isArray(images) && images.length === 1) {
+    content = {media: images[0]}
+  }
+  // multiple images are special case
```
Comment (janhalama) [inline grid/social-media/publish-post/maps/linkedin.suma:29]: 👍  there is test for this new case

**Review by janhalama — APPROVED** [review]
LGTM, this is huge amount of work 👍

**Pull request merged**